sdk/scheduler/src/test/java/com/mesosphere/sdk/framework/ApiServerTest.java

```diff
@@ -19,36 +30,122 @@
-            Thread.sleep(SHORT_TIMEOUT_MILLIS);
+    private static void checkEndpoints(URI server, Map<String, String> endpoints) throws Exception {
+        HttpHost host = new HttpHost(server.getHost(), server.getPort());
+        for (Map.Entry<String, String> entry : endpoints.entrySet()) {
+            HttpResponse response = HttpClientBuilder.create().build()
+                    .execute(host, new BasicHttpRequest("GET", entry.getKey()));
+            if (entry.getValue() == null) {
+                // Verify return value 404 only
+                Assert.assertEquals(entry.getKey() + ": " + endpoints.toString(),
+                        404, response.getStatusLine().getStatusCode());
+            } else if (entry.getValue().isEmpty()) {
+                // Verify return value 200 only
+                Assert.assertEquals(entry.getKey() + ": " + endpoints.toString(),
+                        200, response.getStatusLine().getStatusCode());
+            } else {
+                // Verify content and return value
+                Assert.assertEquals(entry.getKey() + ": " + endpoints.toString(), entry.getValue(), new ContentResponseHandler()
+                        .handleEntity(response.getEntity())
+                        .asString());
+                Assert.assertEquals(entry.getKey() + ": " + endpoints.toString(),
+                        200, response.getStatusLine().getStatusCode());
+            }
         }
-        Assert.assertTrue(bool.get());
     }
 
-    private static class Listener extends AbstractLifeCycle.AbstractLifeCycleListener {
-
+    private static class Listener implements Runnable {
         private final AtomicBoolean apiServerStarted = new AtomicBoolean(false);
 
         @Override
-        public void lifeCycleStarted(LifeCycle event) {
+        public void run() {
             apiServerStarted.set(true);
         }
+
+        private void waitForStarted() {
+            int maxSleepCount = LONG_TIMEOUT_MILLIS / SHORT_TIMEOUT_MILLIS;
+            for (int i = 0; i < maxSleepCount && !apiServerStarted.get(); ++i) {
+                try {
+                    Thread.sleep(SHORT_TIMEOUT_MILLIS);
+                } catch (Exception e) {
+                    // ignore and continue
+                }
+            }
+            Assert.assertTrue(apiServerStarted.get());
+
+        }
+    }
+
+    @Path("/v1/plans")
+    public static class TestResourcePlans {
+
+        @Path("{planName}")
+        @GET
+        public Response getPlan(@PathParam("planName") String planName) {
+            return Response.ok().entity("Service Plan: " + planName).build();
+        }
+    }
+
+    @Path("/v1/pod")
+    public static class TestResourcePod {
+
+        @Path("/{name}/info")
+        @GET
+        public Response getPlan(@PathParam("name") String podName) {
+            return Response.ok().entity("Service Pod: " + podName).build();
+        }
+    }
+
+    @Path("/v1/jobs")
```

Comment:
If these `jobs` endpoints should land on this side of the SDK vs. Queue divide (I think they do), let's call them `services` instead of `jobs` to align with what the SDK thinks instead of what Queue thinks.  A queue may consider a service to be a job, but that's its business.

Reply:
Yes they're under e.g. `/v1/service/<name>/plans/deploy` in the 'real' version to be upstreamed soon. This test was originally written against an early pass. One catch is this means the full URL would look like `http://cluster.com/service/queue/v1/service/job/plans/deploy`